@xen-orchestra/lite/src/components/Tree.tsx

```diff
@@ -67,16 +71,27 @@ interface Effects {
 interface Computed {}
 
 // Inspired by https://mui.com/components/tree-view/#contentcomponent-prop.
-const CustomContent = React.forwardRef(function CustomContent(props, ref) {
-  const { classes, className, label, nodeId, expansionIcon } = props
+const CustomContent = React.forwardRef(function CustomContent(props: CustomContentProps, ref) {
+  const { classes, className, label, expansionIcon, nodeId, to } = props
   const { handleExpansion, handleSelection, selected } = useTreeItem(nodeId)
+  const history = useHistory()
+
+  const handleSelectionClick = (event: React.SyntheticEvent) => {
+    to !== undefined && history.push(to)
+    handleSelection(event)
+  }
 
   return (
-    <span className={classNames(className, { [classes.selected]: selected })} ref={ref}>
+    <span
+      className={classNames(className, { [classes.selected]: selected })}
+      // To trigger click event in the front of 'label' if there is no 'expansionIcon'.
+      onClick={expansionIcon === undefined ? handleSelectionClick : undefined}
+      ref={ref}
+    >
       <span className={classes.iconContainer} onClick={handleExpansion}>
         {expansionIcon}
       </span>
-      <span className={classes.label} onClick={handleSelection}>
+      <span className={classes.label} onClick={handleSelectionClick}>
         {label}
       </span>
     </span>
```

Comment:
https://github.com/vatesfr/xen-orchestra/pull/6005#discussion_r757540658
`Event#stopPropagation` takes care of that:
```js
  const handleSelectionClick = (event: React.SyntheticEvent) => {
    to !== undefined && history.push(to)
    handleSelection(event)
  }

  const handleExpansionClick = (event: React.SyntheticEvent) => {
    event.stopPropagation()
    handleExpansion(event)
  }

  return (
    <span className={classNames(className, { [classes.selected]: selected })} onClick={handleSelectionClick} ref={ref}>
      <span className={classes.iconContainer} onClick={handleExpansionClick}>
        {expansionIcon}
      </span>
      <span className={classes.label}>{label}</span>
    </span>
  )
```